src/Rules/Functions/PrintfParametersRule.php

```diff
@@ -116,7 +116,7 @@ private function getPlaceholdersCount(string $functionName, string $format): int
 
 		$specifiers = sprintf($specifiers, $addSpecifier);
 
-		$pattern = '~(?<before>%*)%(?:(?<position>\d+)\$)?[-+]?(?:[ 0]|(?:\'[^%]))?-?\d*(?:\.\d*)?' . $specifiers . '~';
+		$pattern = '~(?<before>%*)%(?:(?<position>\d+)\$)?[-+]?(?:(?:[ 0]|(?:\'[^%]))(?<width>\*)?)?-?\d*(?:\.\d*)?' . $specifiers . '~';
```

Comment:
Looking at this regex, I see the precision doesn't allow for a `*` either.

Sorry for not testing that case before opening the issue :confounded: 

From the [doc](https://www.php.net/manual/en/function.printf):

> Note: If the period is specified without an explicit value for precision, 0 is assumed. If * is used, the precision is supplied as an additional integer value preceding the one formatted by the specifier.